oioioi/su/views.py

```diff
@@ -38,6 +41,10 @@ def su_view(request, next_page=None, redirect_field_name=REDIRECT_FIELD_NAME):
     user = form.cleaned_data['user']
     if is_under_su(request):
         raise SuspiciousOperation
+    if not is_superuser(request) and not Participant.objects.filter(
+        user=user, contest=request.contest
+    ).exists():
+        raise SuspiciousOperation
```

Comment:
Basicadmins shouldn't be allowed to su into contest admins/owners (and contest admins into owners).
The check for a Participant isn't sufficient because a contest admin could add an owner as a participant or the admins could just be participants, which does occur on my deployments.
This should also be addressed in `get_suable_users_view`.